csrc/ir_internal_nodes.h

```diff
@@ -115,13 +115,19 @@ class TORCH_CUDA_CU_API IndexSelectOp : public Expr {
 class TORCH_CUDA_CU_API TorchGatherOp : public Expr {
  public:
   using Expr::Expr;
+
+  //! torch.gather and numpy.take_along_axis mostly have the same
+  //! semantics, except for that in torch.gather the extents of the
+  //!  non-indexed dimensions of the index tensor may have smaller
+  //! extents than those of the input tensor
   TorchGatherOp(
       IrBuilderPasskey,
       Val* out,
       Val* in,
       int dim,
       IterDomain* select_id,
-      Val* index);
+      Val* index,
+      bool is_take_along_axis);
```

Comment:
Just some comment about naming: should we name the op as `TakeAlongAxis` or `TakeAlongDim`(the latter is the PyTorch name), and rename this parameter as `exact_size`? I don't like the name of `TorchGather` vs `Gather`.

Reply:
That is fine. Name cleanup is low priority and doesn't have to be in this PR. Could you create an issue to track?

Reply:
#214

Reply:
The parameter is renamed to `exact_sizes`